src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/WindowBackdropManager.cs

```diff
@@ -0,0 +1,147 @@
+            WindowBackdropType.TransientWindow => Utility.IsOSWindows7OrNewer,
+            WindowBackdropType.None => true,
+            _ => false
+        };
+    }
+
+    internal static bool SetBackdrop(Window window, WindowBackdropType backdropType)
+    {
+        if (window is null || 
+                !IsSupported(backdropType) || 
+                window.AllowsTransparency ||
+                IsBackdropEnabled == false)
+        {
+            return false;
+        }
+
+        var handle = new WindowInteropHelper(window).Handle;
+        if(handle == IntPtr.Zero)
+        {
+            return false;
+        }
+
+        return SetBackdropCore(handle, backdropType);
+    }
+
+    #region Private Methods
+
+    private static bool SetBackdropCore(IntPtr hwnd, WindowBackdropType backdropType)
+    {
+        ArgumentNullException.ThrowIfNull(hwnd, nameof(hwnd));
+
+        if (backdropType == WindowBackdropType.None)
+        {
+            RestoreBackground(hwnd);
+            return RemoveBackdrop(hwnd);
+        }
+
+        RemoveBackground(hwnd);
+        return ApplyBackdrop(hwnd, backdropType);
+    }
+
+    private static bool ApplyBackdrop(IntPtr hwnd, WindowBackdropType backdropType)
+    {
+        UpdateGlassFrame(hwnd, backdropType);
+        
+        var backdropPvAttribute = backdropType switch
+        {
+            WindowBackdropType.Auto => Standard.DWMSBT.DWMSBT_TABBEDWINDOW,
+            WindowBackdropType.TabbedWindow => Standard.DWMSBT.DWMSBT_TABBEDWINDOW,
+            WindowBackdropType.MainWindow => Standard.DWMSBT.DWMSBT_MAINWINDOW,
+            WindowBackdropType.TransientWindow => Standard.DWMSBT.DWMSBT_TRANSIENTWINDOW,
+            _ => Standard.DWMSBT.DWMSBT_NONE
+        };
+
+        var dwmResult = NativeMethods.DwmSetWindowAttributeSystemBackdropType(hwnd, backdropPvAttribute);
+        return dwmResult == HRESULT.S_OK;
+    }
+
+    private static bool RemoveBackdrop(IntPtr hWnd)
+    {
+        UpdateGlassFrame(hWnd, WindowBackdropType.None);
+
+        var backdropPvAttribute = Standard.DWMSBT.DWMSBT_NONE;
+        var dwmResult = NativeMethods.DwmSetWindowAttributeSystemBackdropType(hWnd, backdropPvAttribute);
+        return dwmResult == HRESULT.S_OK;
+    }
+
+    private static bool RemoveBackground(IntPtr hwnd)
+    {
+        var windowSource = HwndSource.FromHwnd(hwnd);
+        if(windowSource?.Handle != IntPtr.Zero && windowSource.CompositionTarget != null)
+        {
+            windowSource.CompositionTarget.BackgroundColor = Colors.Transparent;
+            return true;
+        }
+        return false;
+    }
+
+    private static bool RestoreBackground(IntPtr hwnd)
+    {
+        var windowSource = HwndSource.FromHwnd(hwnd);
+        if(windowSource?.Handle != IntPtr.Zero && windowSource.CompositionTarget != null)
+        {
+            windowSource.CompositionTarget.BackgroundColor = SystemColors.WindowColor;
```

Comment:
Probably not for the demo now, but I would suggest you store the current color somewhere when removing background and return the same color it had when restoring it.

Reply:
Adding a comment here for next iteration.